core-bundle/src/Resources/contao/modules/Module.php

```diff
@@ -551,9 +551,9 @@ protected static function getPublishedSubpagesWithoutGuestsByPid($intPid, $blnSh
 		return array_filter(array_map(
 			static function (array $row) use ($blnFeUserLoggedIn): ?array
 			{
-				$page = PageModel::findWithDetails($row['id']);
+				$page = PageModel::findByPk($row['id']);
 
-				if ($blnFeUserLoggedIn && $page->protected && \in_array(-1, StringUtil::deserialize($page->groups)))
+				if (!$blnFeUserLoggedIn && $page->loadDetails()->protected && \in_array(-1, StringUtil::deserialize($page->groups)))
```

Comment:
```suggestion
				if ($blnFeUserLoggedIn && $page->loadDetails()->protected && \in_array(-1, StringUtil::deserialize($page->groups)))
```

The page should only be skipped **if** a user is logged in and the page is only for guests I think.